File: execution-engine/test_engine.py
import requests

url = 'http://localhost:8080'

print('--- Pinging Server ---')
try:
    print('Ping response:', requests.get(url + '/ping').text)
except Exception as e:
    print('Failed to reach server:', e)

print('\n--- Testing Python Execution ---')
p_payload = {'language': 'python', 'code': 'print("Hello Python")'}
try:
    res = requests.post(url + '/execute', json=p_payload)
    print('Status:', res.status_code, 'Response:', res.json())
except Exception as e:
    print('Error:', e)

print('\n--- Testing C++ Execution ---')
c_payload = {'language': 'cpp', 'code': '#include <iostream>\nint main() { std::cout << "Hello C++" << std::endl; return 0; }'}
try:
    res = requests.post(url + '/execute', json=c_payload)
    print('Status:', res.status_code, 'Response:', res.json())
except Exception as e:
    print('Error:', e)

print('\n--- Testing Python Timeout ---')
err_payload = {'language': 'python', 'code': 'while True: pass'}
try:
    res = requests.post(url + '/execute', json=err_payload)
    print('Status:', res.status_code, 'Response:', res.json())
except Exception as e:
    print('Error:', e)

File: execution-engine/test_engine.js
const http = require('http');

async function sendRequest(path, data = null) {
  return new Promise((resolve, reject) => {
    const options = {
      port: 8080,
      path: path,
      method: data ? 'POST' : 'GET',
      headers: {
        'Content-Type': 'application/json'
      }
    };

    const req = http.request(options, (res) => {
      let result = '';
      res.on('data', chunk => result += chunk);
      res.on('end', () => resolve({ status: res.statusCode, data: result }));
    });

    req.on('error', e => reject(e));

    if (data) req.write(JSON.stringify(data));
    req.end();
  });
}

(async () => {
  try {
    console.log('--- Pinging Server ---');
    console.log(await sendRequest('/ping'));

    console.log('\n--- Testing Python Execution ---');
    console.log(await sendRequest('/execute', { language: 'python', code: 'print("Hello Python")' }));

    console.log('\n--- Testing C++ Execution ---');
    console.log(await sendRequest('/execute', { language: 'cpp', code: '#include <iostream>\nint main() { std::cout << "Hello C++" << std::endl; return 0; }' }));

    console.log('\n--- Testing Python Timeout ---');
    const start = Date.now();
    console.log(await sendRequest('/execute', { language: 'python', code: 'while True: pass' }));
    console.log(`Timeout test took ${Date.now() - start}ms`);

  } catch (err) {
    console.error(err);
  }
})();
